lib/determinator/retrieve/http_retriever.rb

```diff
@@ -10,11 +10,14 @@ def initialize(connection:)
       end
 
       def retrieve(name)
+        before_hook
         response = @connection.get("/features/#{name}")
+        after_hook(response.status)
```

Comment:
As discussed, can we pass in the error here? (nil in this case, and populating it inside the rescue).

Reply:
done